### PR TITLE
Origin/refactor/setting correct types

### PR DESCRIPTION
**Pull request by Banshaan**
Fixed the getViews method so that the Observable is typed.

### CODE REVIEW

File: src/app/shared/services/http.service.ts [hunk 1]
```diff
@@ -31,8 +32,8 @@ export class HttpService {
     this.toastService.showSuccess(message);
   }
 
-  getViews(): Observable<any> {
-    return this.http.get('api/testtool/views').pipe(catchError(this.handleError()));
+  getViews(): Observable<{ [key: string]: View }> {
+    return this.http.get<{ [key: string]: View }>('api/testtool/views').pipe(catchError(this.handleError()));
```
Comment (philipsens) [inline src/app/shared/services/http.service.ts:36]: Hi, thanks for the PR. 

Could you change `{ [key: string]: View }` to `Record<string, View>` that reads a bit better?

Comment (mhdirkse) [reply src/app/shared/services/http.service.ts:36]: Or Map<string, View>?. Wouldn't that be better, Sergi?

Comment (philipsens) [reply src/app/shared/services/http.service.ts:36]: Map does more than what @Banshaan needs it for in this case. `Record` is a Utility type: https://www.typescriptlang.org/docs/handbook/utility-types.html

Comment (Banshaan) [reply src/app/shared/services/http.service.ts]: Changed { [key: string]: View } to Record<string, View>

**Review by philipsens — APPROVED** [review]

**Pull request merged**